psqlgraph/node.py

```diff
@@ -109,6 +111,51 @@ def get_edges(self):
         for edge_out in self.edges_out:
             yield edge_out
 
+    def bfs_children(self, edge_predicate=None, transform=None):
+        """
+        Perform a BFS, with `self` being the root node and perform arbitrary
+        node transformation along the way
+
+        :param edge_predicate: a predicate performed on an `edge` object in
+            order to decided whether to walk that edge or not
+        :type edge_predicate: func
+        :param transform: a transformation that needs to be done on the node
+        :type transform: func
+
+        :return: List[str]
+        """
+
+        if not callable(edge_predicate):
+            def edge_predicate(e):
+                return True
+
+        if not callable(transform):
+            def transform(node):
+                return node
+
+        traversal = []
+        visited = set()
+        queue = deque([self])
+
+        visited.add(self.node_id)
+
+        while queue:
+            current = queue.popleft()
+            transform(current)
```

Comment:
yeah, @khan08 and @kulgan suggested a `generator` too, so my idea was to either:
* transform is part of this and I also return the `traversal`, because we might need to know what has been changed 
* return a `generator` and transform is outside of this method
so I went with the first approach.

Reply:
but yeah, I'm fine with `generator`, since it's more efficient.